deps/wazuh_testing/wazuh_testing/scripts/statistical_data_analyzer.py

```diff
@@ -0,0 +1,339 @@
+        value: metric on which the test is performed.
+    
+    Returns:
+        l_p_value: p value returned by the Levene test.
+    """
+    ref_values = baseline[value].dropna()
+    new_values = datasource[value].dropna() 
+
+    l_stat, l_p_value = levene(ref_values, new_values)
+
+    return l_p_value
+
+
+def t_anova_test(baseline, datasource, value):
+    """Function that performs the statistical analysis using the ANOVA test.
+
+    Args:
+        baseline: Dataframe with the baseline data.
+        datasource: Dataframe with the data source.
+        value: metric on which the test is performed.
+
+    Returns:
+        a_p_value: p value returned by the ANOVA test.
+    """
+    ref_values = baseline[value].dropna()
+    new_values = datasource[value].dropna() 
+
+    a_stat, a_p_value = f_oneway(ref_values, new_values)
+
+    return a_p_value
+
+
+def data_comparison_test(baseline, datasource, percentage=95):
+    """Calculate statistics values for t-Student, Levene tests and ANOVA test, and shows the results. 
+    Also, it detect if there are significant difference detected between both values.
+
+    Args:
+        baseline: file with the baseline values.
+        datasource: file with the values to compare.
+        percentage: percentage of confidence level.
+    """
+    dataframe_ref = load_dataframe(baseline)
+    dataframe = load_dataframe(datasource)
+    alpha = float((100 - percentage) / 100)
+    daemons = dataframe_ref['Daemon'].unique()
+
+    for daemon in daemons:
+        print(f"\nDaemon: {daemon}\n")
+        ref_daemon_data = dataframe_ref[dataframe_ref['Daemon'] == daemon]
+        new_daemon_data = dataframe[dataframe['Daemon'] == daemon]
+
+        for value in STATS_MAPPING.values():
+            ref_values = ref_daemon_data[value].dropna()
+            new_values = new_daemon_data[value].dropna()
+
+            # t-Student
+            t_stat, t_p_value = ttest_ind(ref_values, new_values, equal_var=False)
+            # Levene
+            l_stat, l_p_value = levene(ref_values, new_values)
+            # Anova
+            a_stat, a_p_value = f_oneway(ref_values, new_values)
+
+            print(f"\nMetric '{value}':")
+            print(f"    - t-Student Test: t-Statistic={t_stat}, p-value={t_p_value}")
+            print(f"    - Levene's Test: Statistic={l_stat}, p-value={l_p_value}")
+            print(f"    - ANOVA's Test: Statistic={a_stat}, p-value={a_p_value}")
+
+            if t_p_value < alpha:
+                print(f"\n  Significant difference detected in '{value}' with {percentage}% confidence (t-Test).")
+            if l_p_value < alpha:
+                print(f"\n  Variance difference detected in '{value}' with {percentage}% confidence (Levene's Test).")
+            if a_p_value < alpha:
+                print(f"\n  Means difference detected in '{value}' with {percentage}% confidence (ANOVA's Test).")
+        print("\n")
+
+
+def print_dataframes_stats(baseline, datasource):
```

Comment:
Same, these functions can be a good addition to a test or an external tool to perform analysis, but not in the library.

Reply:
This function generates comparative tables between the statistics of both data sets and adds them to the final report, so its use is necessary. Likewise, its design has been modified in https://github.com/wazuh/wazuh-qa/pull/5622/commits/26dc5cfa8d7f2339d13d5d5820d1ad7cceda3625